presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -339,10 +342,24 @@ private PlanBuilder explicitCoercionSymbols(PlanBuilder subPlan, Iterable<Symbol
 
     private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
     {
-        if (analysis.getAggregates(node).isEmpty() && analysis.getGroupByExpressions(node).isEmpty()) {
+        if (analysis.getAggregates(node).isEmpty() && analysis.getGroupingSets(node).isEmpty()) {
             return subPlan;
         }
 
+        List<List<FieldOrExpression>> groupingSets = analysis.getGroupingSets(node);
+        if (groupingSets.size() == 0) {
```

Comment:
This decision should probably be made in the analyzer. If there are aggregates but no group by clause, it add an empty grouping set to the analysis result.

Then, this code would just blindly call `aggregateSingleGroupingSet` with the single element that it expects.